packages/light-client-experimental/src/smoldot/resources/chain.ts

```diff
@@ -0,0 +1,31 @@
+import * as bindings from "../bindings"
+import * as smoldot from "smoldot"
+
+import { Cause, Effect, pipe as $ } from "effect"
+import { AddChainError, AlreadyDestroyedError, CrashError } from "../data"
+
+export type AcquireOptions = smoldot.AddChainOptions & {
+  client: bindings.client.Client
+}
+
+export const acquire = (
```

Comment:
I'd like to see more comments around this code, eg what is the intended purpose of these `acquire` and `release` functions?